### PR TITLE
Bump hyper to 0.14.3 and some best practises

### PR DESCRIPTION
**Pull request by karim-agha**
- reacting to: https://rustsec.org/advisories/RUSTSEC-2021-0020.html
- additional code cleanup to use `serde` instead of `serde_derive`.
- additional code cleanup to remove deprecated pre-rust2018 `extern crate`.
- updated some other crate versions.

### CODE REVIEW

File: examples/path/globs/Cargo.toml [hunk 1]
```diff
@@ -11,5 +11,4 @@ gotham = { path = "../../../gotham" }
 gotham_derive = { path = "../../../gotham_derive" }
 
 mime = "0.3"
-serde = "1"
-serde_derive = "1"
+serde = { version = "1.0", features = ["derive"] }
```
Comment (msrd0) [inline examples/path/globs/Cargo.toml:14]: Please ensure all files end with a newline

**Review by msrd0 — CHANGES_REQUESTED** [review]
Thanks for applying some best practises! The CI has some failed tests, can you fix those please?

**Comment by msrd0** [comment]
Hyper has been bumped in #524 already, but I'd still be interested in merging the other changes you made.

**Comment by karim-agha** [comment]
I can't repro the clippy errors locally

**Pull request closed without merging**